src/main/kotlin/app/revanced/patches/remini/misc/premium/patch/PremiumUnlockPatch.kt

```diff
@@ -0,0 +1,38 @@
+package app.revanced.patches.remini.misc.premium.patch
+
+import app.revanced.patcher.annotation.Description
+import app.revanced.patcher.annotation.Name
+import app.revanced.patcher.annotation.Version
+import app.revanced.patcher.data.BytecodeContext
+import app.revanced.patcher.extensions.MethodFingerprintExtensions.name
+import app.revanced.patcher.extensions.replaceInstruction
+import app.revanced.patcher.patch.BytecodePatch
+import app.revanced.patcher.patch.PatchResult
+import app.revanced.patcher.patch.PatchResultError
+import app.revanced.patcher.patch.PatchResultSuccess
+import app.revanced.patcher.patch.annotations.Patch
+import app.revanced.patches.remini.misc.premium.annotations.PremiumUnlockCompatibility
+import app.revanced.patches.remini.misc.premium.fingerprints.PremiumUnlockFingerprint
+
+@Patch
+@Name("premium-unlock")
+@Description("Unlocks premium-only functions.")
+@PremiumUnlockCompatibility
+@Version("0.0.1")
+class PremiumUnlockPatch : BytecodePatch(
+    listOf(PremiumUnlockFingerprint)
+) {
+    override fun execute(context: BytecodeContext): PatchResult {
+        val result = PremiumUnlockFingerprint.result ?: return PatchResultError("${PremiumUnlockFingerprint.name} not found")
```

Comment:
Use the extenson method `MethodFingerprint.toPatchResultError()` here instead of creating the return value yourself.